flocker/node/agents/cinder.py

```diff
@@ -131,17 +191,20 @@ def create_volume(self, dataset_id, size):
         }
         # We supply metadata here and it'll be included in the returned cinder
         # volume record, but it'll be lost by Rackspace, so...
-        requested_volume = self.volume_manager.create(
+        requested_volume = self.cinder_volume_manager.create(
             size=Byte(size).to_GB().value,
             metadata=metadata,
         )
-        created_volume = wait_for_volume(self.volume_manager, requested_volume)
+        created_volume = wait_for_volume(self.cinder_volume_manager, requested_volume)
         # So once the volume has actually been created, we set the metadata
         # again. One day we hope this won't be necessary.
         # See Rackspace support ticket: 150422-ord-0000495'
-        self.volume_manager.set_metadata(created_volume, metadata)
+        self.cinder_volume_manager.set_metadata(created_volume, metadata)
```

Comment:
Only the Cinder API is capable of setting volume metadata, so we need it as well as the Nova API.